### PR TITLE
Allow any file extension for README file

### PR DESCRIPTION
**Pull request by s-spindler**
# Pull Request Check List

Resolves: #5356

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

File: docs/cli.md [hunk 1]
```diff
@@ -101,7 +101,7 @@ my-package
 
 * `--name`: Set the resulting package name.
 * `--src`: Use the src layout for the project.
-* `--readme`: Specify the readme file format. One of `md` (default) or `rst`.
+* `--readme`: Specify the readme file format. Default is `md`.
```
Comment (neersighted) [inline docs/cli.md:104]:
```suggestion
* `--readme`: Specify the readme file extension. Default is `md`.
```

**Review by neersighted — CHANGES_REQUESTED** [review]
I'm fine with this -- looks like it just needs a little touchup and it's good.

**Comment by Secrus** [comment]
I think it would be good to add a warning or note in `pyproject` docs, that formats different from md or rst might not be processed correctly on official PyPI.

**Comment by s-spindler** [comment]
I added @neersighted's suggestion and added a sentence as suggested by @Secrus.

**Review by neersighted — DISMISSED** [review]

**Comment by Secrus** [comment]
@s-spindler please fix the pre-commit problem so we could merge this.

**Review by Secrus — APPROVED** [review]

**Pull request merged**